server/app/repository/ProgramRepository.java

```diff
@@ -161,7 +162,21 @@ public void setFullProgramDefinitionCache(long programId, ProgramDefinition prog
         && getFullProgramDefinitionFromCache(programId).isEmpty()) {
       // We should never set the cache for draft programs.
       if (!versionRepository.get().isDraftProgram(programId)) {
-        programDefCache.set(String.valueOf(programId), programDefinition);
+        ImmutableList<BlockDefinition> blocksWithNullQuestion =
+            programDefinition.blockDefinitions().stream()
+                .filter(BlockDefinition::hasNullQuestion)
+                .collect(ImmutableList.toImmutableList());
+        if (blocksWithNullQuestion.size() > 0) {
+          logger.warn(
```

Comment:
Can you flatten the `blocksWithNullQuestion` and include a csv of the questionIds with the log statement?

Reply:
Done! Did it within the function for now, but let me know if you think it's worth pulling out into something separate.